### PR TITLE
Bug Fixes

### PR DESCRIPTION
**Pull request by AbdullahAhmedSoomro**
Bug Fixes
Issues that were mentioned in these PRs
- Feature Firebase Analytics (PR #522 ).
- Feature Firebase Remote Config (PR #521 ).

Pending:

- Setup Injector for Firebase  #524

### CODE REVIEW

**Comment by AbdullahAhmedSoomro** [comment]
Bug Fixes

- TextField hide when keyboard opens issue fixed.
- Setup Injector for Firebase (Issue #524)
- Get User Name Function from Backend. (Issue #471)
- Grey screen appearing when clicking on View Details (Issue #462)
- Bug/ProfileScreen (Issue #458)
- UI/ [BUG] Status Bar Color (Issue #409)

File: lib/main.dart [hunk 1]
```diff
@@ -21,48 +20,42 @@ String createDirectory({required String path}) {
...
-    final service = FirebaseService();
-    await service.initialise();
-    final reportingService = ReportingService();
-    await reportingService.initialise();
-    final analyticsService = AnalyticsService();
-    await analyticsService.initialise();
+  /// Firebase Services initialisation
+  await Injector.resolve<FirebaseService>().initialise();
+  await Injector.resolve<ReportingService>().initialise();
+  await Injector.resolve<AnalyticsService>().initialise();
```
Comment (hashirshoaeb) [inline lib/main.dart:36]: Can we initialise them in parallel?

File: lib/presentation/injector/injector.dart [hunk 2]
```diff
@@ -46,7 +50,25 @@ class _Injector extends Injector {
     final KiwiContainer container = KiwiContainer();
     final SharedPreferences preferences = await SharedPreferences.getInstance();
     container
-        .registerSingleton<AppSettings>((c) => AppSettingsImpl(preferences));
+      ..registerSingleton<AppSettings>((c) => AppSettingsImpl(preferences))
+      ..registerSingleton<FirebaseService>((c) {
+        final service = FirebaseService();
+        print('initialising firebase service');
+        service.initialise();
```
Comment (hashirshoaeb) [inline lib/presentation/injector/injector.dart:57]: initialising twice?

**Review by hashirshoaeb — APPROVED** [review]

File: lib/presentation/injector/injector.dart (hunk 2 above)
Comment (AbdullahAhmedSoomro) [reply lib/presentation/injector/injector.dart:57]: Oops I have recheck this I think i missed it

**Review by hashirshoaeb — APPROVED** [review]

**Pull request merged**